cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go

```diff
@@ -65,6 +65,10 @@ type InitConfiguration struct {
 	// "kubeadm init".
 	// +optional
 	Patches *Patches `json:"patches,omitempty"`
+
+	// Timeout holds timeout fields that relate to registering a new control-plane or node to the cluster,
+	// either via "kubeadm init" or "kubeadm join"
+	Timeouts InitTimeouts `json:"timeout,omitempty"`
```

Comment:
```suggestion
	// +optional
	Timeouts *InitTimeouts `json:"timeout,omitempty"`
```

- should be a pointer.
- have +optional